grpc/stream.go

```diff
@@ -343,12 +344,16 @@ func (s *stream) close(err error) {
 	s.logger.WithError(err).Debug("stream is closing")
 
 	s.state = closed
-	close(s.done)
+
+	s.doneOnce.Do(func() {
+		close(s.done)
+	})
```

Comment:
This should never be needed as this function should only be called on the event loop so synchronized and the check above should prevent any race.

Did you experience a race?

Reply:
You're absolutely right, my fault! For a moment I forgot about the javascript event loop. I'm sorry, I'll undo this!